filament/src/RenderPass.cpp

```diff
@@ -390,17 +391,18 @@ void RenderPass::generateCommandsImpl(uint32_t extraFlags,
          * This is our hot loop. It's written to avoid branches.
          * When modifying this code, always ensure it stays efficient.
          */
-        for (auto const& primitive : primitives) {
+        for (size_t pi = 0, c = primitives.size(); pi < c; ++pi ) {
+            auto const& primitive = primitives[pi];
             FMaterialInstance const* const mi = primitive.getMaterialInstance();
-            FMorphTargetBuffer const* const morphTargetBuffer = primitive.getMorphTargetBuffer();
             if constexpr (isColorPass) {
                 cmdColor.primitive.primitiveHandle = primitive.getHwHandle();
                 cmdColor.primitive.materialVariant = variant;
                 RenderPass::setupColorCommand(cmdColor, mi, inverseFrontFaces);
 
                 cmdColor.primitive.morphWeightBuffer = morphing.handle;
-                if (UTILS_UNLIKELY(morphTargetBuffer)) {
-                    cmdColor.primitive.morphTargetBuffer = morphTargetBuffer->getHwHandle();
+                if (UTILS_UNLIKELY(hasMorphing)) {
+                    auto& morphTargets = morphing.targets[pi];
+                    cmdColor.primitive.morphTargetBuffer = morphTargets.buffer->getHwHandle();
```

Comment:
I was just looking at this yesterday. I really don't like that this adds (added) a branch in our inner hot loop.
Is there a way we can replace this whole `if` by just:

```
cmdColor.primitive.morphTargetBuffer = primitive.getMorphTargetHandle();
```

The RenderableManager should be able to populate this. In other words, could we use FRenderPrimitive has a staging cache for `soaMorphing[i].targets[pi].buffer->getHwHandle();`

Reply:
Ok... I'm done by creating `MorphTargets` always and set dummy `MorphTargetBuffers`. Please review new changes. By these changes, I can remove more branches in hot loops.